src/main/java/com/endava/soa_5ed/task5_io/ex1/PropertiesManager.java

```diff
@@ -0,0 +1,44 @@
+package com.endava.soa_5ed.task5_io.ex1;
+
+import java.io.*;
+import java.util.Properties;
+
+public class PropertiesManager {
+    String result;
+    private Properties properties;
+
+    public PropertiesManager(String filepath) {
+
+        try {
+            properties = readPropFile(filepath);
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+    }
+
+    public Properties readPropFile(String filepath) throws IOException {
+        FileInputStream inputStream = null;
+        Properties prop = null;
+        try {
+            inputStream = new FileInputStream(filepath);
+            prop = new Properties();
+            prop.load(inputStream);
+        } catch (FileNotFoundException fileNotFound) {
+            System.out.println(fileNotFound.getMessage());
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        } finally {
+            assert inputStream != null;
+            inputStream.close();
+        }
+        return prop;
+    }
+
+    public String getProperty(String key) throws NoValueException {
+        result = properties.getProperty(key);
+        if (result.isEmpty()) {
```

Comment:
What happening if it is passed an inexistent key and result will be null ?

Reply:
you'll get a NullPointerException, I've tried handling it but never came to a working solution

Reply:
Added Keys enum to help control keys given as parameters which should solve the mentioned issue

Reply:
In this case a Keys object can be given as a parameter, but can be used methods like Strings.isNullOrEmpty() (depending on your dependencies or can create an own method to check empty and null strings